[#9971] Gracefully error out in ETDump for get_flatbuffer_scalar_type

zhenyan-zhang-meta opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #10056

Following https://github.com/pytorch/executorch/issues/9971

- Update get_flatbuffer_scalar_type return type to Result<T>
- Iteratively update functions that calling the functions with result type changed:
  - Check returns, if with an error, pass above the error.
  - If unable to pass error, update the return type as Result<T>

Differential Revision: [D72771753](https://our.internmc.facebook.com/intern/diff/D72771753/)

Commit:
[#9971] Gracefully error out in ETDump for get_flatbuffer_scalar_type

Following https://github.com/pytorch/executorch/issues/9971

- Update get_flatbuffer_scalar_type return type to Result<T>
- Iteratively update functions that calling the functions with result type changed:
  - Check returns, if with an error, pass above the error.
  - If unable to pass error, update the return type as Result<T>

Differential Revision: [D72771753](https://our.internmc.facebook.com/intern/diff/D72771753/)

[ghstack-poisoned]

Gasoonjia commented on runtime/core/event_tracer.h:
```diff
@@ -314,7 +314,7 @@ class EventTracer {
    * @param[in] evalue_type Indicates what type of output this is logging e.g.
    * an intermediate output, program output etc.
    */
-  virtual void log_evalue(
```
Please update the comment for result value

Gasoonjia reviewed (approved):
LGTM! Thanks for update!
Please update the comment before landing it.

Commit:
Update on "[#9971] Gracefully error out in ETDump for get_flatbuffer_scalar_type"


Following https://github.com/pytorch/executorch/issues/9971

- Update get_flatbuffer_scalar_type return type to Result<T>
- Iteratively update functions that calling the functions with result type changed:
  - Check returns, if with an error, pass above the error.
  - If unable to pass error, update the return type as Result<T>

Differential Revision: [D72771753](https://our.internmc.facebook.com/intern/diff/D72771753/)

[ghstack-poisoned]
